01-algorithms/01-algorithms.md

```diff
@@ -288,42 +288,42 @@ Still feeling stuck? Try watching the video walkthrough of a sample solution bel
 A sample solution could be:
 
 ```py
-  def duplicates_within_k(numbers, k):
-    '''
-    INPUT: list of integers and integer k
-    OUTPUT: Boolean indicating whether there is are duplicate elements in the list within k distance
-    '''
-    # get length of input list
-    lst_length = len(numbers)
-
-    # if the input list has less than two elements or k is zero
-    if lst_length < 2 or k == 0:
-      # return false, there cannot be duplicates
-      return False
-    
-    # loop through indices of list
-    for i in range(lst_length):
-      # initialize a second pointer to track end of subarray
-      j = i + 1
-      # initialize a variable to k
-      # will help make sure our subarray stays within size k + 1
-      dist_remaining = k
-      
-      # while we are within k distance from i
-      # and j is not out of bounds of list
-      while dist_remaining > 0 and j < lst_length:
-        # if elements at index i and j are the same
-        if numbers[i] == numbers[j]:
-          # a duplicate within k distance exists
-          return True
-        # otherwise increment j to increase subarray size
-        j += 1
-        # decrement dist_remaining so subarray will stay within
-        # k distance from i
-        dist_remaining -= 1
-    # if we loop through all subarrays and have not found a duplicate
-    # return False
+def duplicates_within_k(numbers, k):
+  '''
+  INPUT: list of integers and integer k
+  OUTPUT: Boolean indicating whether there is are duplicate elements in the list within k distance
+  '''
+  # get length of input list
+  lst_length = len(numbers)
+
+  # if the input list has less than two elements or k is zero
+  if lst_length < 2 or k == 0:
+    # return false, there cannot be duplicates
     return False
+  
+  # loop through indices of list
+  for i in range(lst_length):
+    # initialize a second pointer to track end of subarray
+    j = i + 1
+    # initialize a variable to k
+    # will help make sure our subarray stays within size k + 1
+    dist_remaining = k
+    
+    # while we are within k distance from i
+    # and j is not out of bounds of list
+    while dist_remaining > 0 and j < lst_length:
+      # if elements at index i and j are the same
+      if numbers[i] == numbers[j]:
+        # a duplicate within k distance exists
+        return True
+      # otherwise increment j to increase subarray size
+      j += 1
+      # decrement dist_remaining so subarray will stay within
+      # k distance from i
+      dist_remaining -= 1
+  # if we loop through all subarrays and have not found a duplicate
+  # return False
```

Comment:
```suggestion
```

I believe this comment can be removed

Reply:
It does read redundant with the following line of code, but it's the end of the sentence starting on the line above. I'll remove it and rephrase so it's less awkward-looking.